server/events/project_command_runner.go

```diff
@@ -114,6 +108,58 @@ type ProjectCommandRunner interface {
 	ProjectApprovePoliciesCommandRunner
 }
 
+// ProjectCommandRunnerWithJobs is a decorator that creates a new PR status check per project.
+// The status contains a url that outputs current progress of the terraform plan/apply command.
+type ProjectCommandRunnerWithJobs struct {
+	ProjectCmdOutputHandler handlers.ProjectCommandOutputHandler
+	ProjectCommandRunner    ProjectCommandRunner
+}
+
+func (p *ProjectCommandRunnerWithJobs) Plan(ctx models.ProjectCommandContext) models.ProjectResult {
+	// Reset the buffer when running the plan. We only need to do this for plan,
+	// apply is a continuation of the same workflow
+	p.ProjectCmdOutputHandler.Clear(ctx)
+	return p.updateProjectPRStatus(models.PlanCommand, ctx, p.ProjectCommandRunner.Plan)
+}
+
+func (p *ProjectCommandRunnerWithJobs) PolicyCheck(ctx models.ProjectCommandContext) models.ProjectResult {
+	return p.ProjectCommandRunner.PolicyCheck(ctx)
+}
+
+func (p *ProjectCommandRunnerWithJobs) ApprovePolicies(ctx models.ProjectCommandContext) models.ProjectResult {
+	return p.ProjectCommandRunner.ApprovePolicies(ctx)
+}
```

Comment:
If you just declare this as:

```go
type ProjectCommandRunnerWithJobs struct {
   ProjectCommandRunner
   ProjectCmdOutputHandler handlers.ProjectCommandOutputHandler
}
```

Then you don't need to declare `PolicyCheck` or `ApprovePolicies` since it will automagically delegate.